tests/test_io.nim

```diff
@@ -0,0 +1,88 @@
+# Stint
+# Copyright 2018 Status Research & Development GmbH
+# Licensed under either of
+#
+#  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE) or http://www.apache.org/licenses/LICENSE-2.0)
+#  * MIT license ([LICENSE-MIT](LICENSE-MIT) or http://opensource.org/licenses/MIT)
+#
+# at your option. This file may not be copied, modified, or distributed except according to those terms.
+
+import ../src/stint, unittest, strutils
+
+suite "Testing input and output procedures":
+  test "Creation from decimal strings":
+    block:
+      let a = parse(Stint[64], "123456789")
+      let b = 123456789.stint(64)
+
+      check: a == b
+      check: 123456789'i64 == cast[int64](a)
+
+    block:
+      let a = parse(Stuint[64], "123456789")
+      let b = 123456789.stuint(64)
+
+      check: a == b
+      check: 123456789'u64 == cast[uint64](a)
+
+    block:
+      let a = parse(Stint[64], "-123456789")
+      let b = (-123456789).stint(64)
+
+      check: a == b
+      check: -123456789'i64 == cast[int64](a)
+
+  test "Creation from hex strings":
+    block:
+      let a = parse(Stint[64], "0xFF", 16)
+      let b = 255.stint(64)
+
+      check: a == b
+      check: 255'i64 == cast[int64](a)
+
+    block:
+      let a = parse(Stuint[64], "0xFF", 16)
+      let b = 255.stuint(64)
+
+      check: a == b
+      check: 255'u64 == cast[uint64](a)
+
+    block:
+      let a = parse(Stint[16], "0xFFFF", 16)
```

Comment:
After investigation, one or more of the noInit wasn't correct, for now I've turned them all to no-op (fooPragma). I will reactivate them slowly until I catch the culprit.

Reply:
Regarding the parsing tests, after the base is done I plan to add more to quickcheck.

For big ints arithmetics I can add a few manual cases but it would be best to have randomized testing against a reference implementation as well. I checked a bit nim-bigints but our test suite is already more comprehensive so we actually might catch nim-bigints bugs :D.